tracee-ebpf/tracee/broker/broker.go

```diff
@@ -0,0 +1,64 @@
+package broker
```

Comment:
this broker design creates a (loose) coupling from tracee-ebpf to the consumers, a coupling which did not exist until now. The coupling is demonstrated at the function call from the broker (broker is a tracee-ebpf concept) to the registered consumer. a function call in this direction is not ideal because it's synchronously called within the critical path of event processing. If one time once streamer is stuck for one event, it could clog the entire pipeline. I would rather keep tracee-ebpf oblivious to it's consumers. To do this, the broker should be a simpler "fan out" machine, that duplicates the incoming event into consumer-specific channels. An example of this pattern can be found in tracee-rules' engine (dispatching code), here's a simpler example: https://stackoverflow.com/a/36418082/310298

Reply:
we can just `go streamer.Stream()` without having a blocking operation. 
Copying here is a bad idea in my opinion.

Reply:
so you suggest to change the broker code to invoke the callback in a goroutine? this will create a whole lot of different challenges, without seeing the code, I can expect ordering issues and resource contention for starts

Reply:
Yes, you are right. Let's go with the "fan out" machine.